source/dynamical_systems/include/dynamical_systems/DynamicalSystemFactory.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "dynamical_systems/IDynamicalSystem.hpp"
+
+namespace dynamical_systems {
+
+/**
+ * @class DynamicalSystemFactory
+ * @brief Factory to create a shared pointer to a dynamical system.
+ * @tparam S Underlying state type of the dynamical system
+ */
+template<class S>
+class DynamicalSystemFactory {
+public:
+  /**
+   * @brief Enumeration of the implemented dynamical systems
+   */
+  enum DYNAMICAL_SYSTEM {
```

Comment:
I believe a `class enum` would be preferred here (see https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum for the reasons)